services/venice-router/src/main/java/com/linkedin/venice/router/streaming/VeniceChunkedWriteHandler.java

```diff
@@ -67,6 +67,10 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
         return;
       }
     }
+    // check for closed channel
+    if (!ctx.channel().isActive() || !ctx.channel().isWritable()) {
+      return;
```

Comment:
Returning early can also result in memory leaking since the following logic will catch all the exception and decrease the reference count and complete `promise`.
If we want to reduce the exception logging, I think we should still log this behavior and in the meantime, release the message and fail the `promise` before returning.